File: ComputeF.m
function [Fn1]=ComputeF(V,x,nabla_W0,N,Xold)
Fn1=zeros(2,2,N);
nabla_W_tmp=[0,0];
Fn1_tmp=zeros(2,2);

for i=1:N
    for j=1:N
        for beta=1:2
            for alpha=1:2
                uj=x(alpha,j)-Xold(alpha,j);
                ui=x(alpha,i)-Xold(alpha,i);
 Fn1_tmp(alpha,beta)= Fn1_tmp(alpha,beta)+(V(j)* (uj-ui) *nabla_W0(beta,i,j));  
             end
        end
    end
    Fn1(1:2,1:2,i)=Fn1_tmp+eye(2);
    Fn1_tmp=zeros(2,2);
end
    
             

File: Visco_plasticity/Visco_plasticity/Compute_F.f90
subroutine Compute_F(vol,x,x_old,nabla_W_0,N,F,table)
    
    integer :: N
    
    real*8 :: vol(N)
    real*8 :: x(2,N)
    real*8 :: x_old(2,N)
    
    real*8 :: nabla_W_0(2,N,N)
    real*8 :: F(2,2,N)
    real*8 ui,uj
    integer :: table(N,120)
    F=0
        do i=1,N
            do j=1,table(i,1)
                do beta=1,2
                    do alpha=1,2
                        uj=x(alpha,table(i,j+1))-x_old(alpha,table(i,j+1))
                        ui=x(alpha,i)-x_old(alpha,i)
                        F(alpha,beta,i)=F(alpha,beta,i)+vol(table(i,j+1))*(uj-ui)*nabla_W_0(beta,i,table(i,j+1))
                    enddo
                enddo
            enddo
            F(1,1,i)= F(1,1,i)+1.0d0
            F(2,2,i)= F(2,2,i)+1.0d0
        enddo
        
    return
end